src/requirements/Uniswap/hooks/useParseVaultAddress.ts

```diff
@@ -0,0 +1,20 @@
+import { useMemo } from "react"
+import { useFormContext, useWatch } from "react-hook-form"
+
+// Note: We might want to put these kind of constants to `@guildxyz/types/consts`
+export const ADDRESS_REGEX = /0x[0-9a-f]{40}/i
+
+export function useParseVaultAddress(baseFieldPath: string) {
+  const { setValue } = useFormContext()
+  const lpVaultInputValue = useWatch({ name: `${baseFieldPath}.data.lpVault` })
```

Comment:
Just to make sure if I understand:
So we use this field client side only, right? And this won't cause any issues when submitting the requirement to our API, nor when we parse it on gate, since we use Zod schemas, and those will just remove the `data.lpVault` field? 👀 Clever!

Reply:
Exactly! The BE's handler won't see this :ayy:

Reply:
Amazing! 💯